File: Stacks/Easy/Valid-Parentheses.java
package Stacks.Easy;
import java.util.Stack;

class Solution {
  public boolean isValid(String s) {
      Stack<Character> stack = new Stack<Character>();
      
      if (s.length() <= 1) {
          return false;
      }
      
      for (int i = 0; i < s.length(); i++) {
          if (s.charAt(i) =='{' || s.charAt(i) == '[' || s.charAt(i) == '(') {
              stack.push(s.charAt(i));
              
          } else if (!stack.isEmpty() && s.charAt(i) == '}' && stack.peek() == '{') {
              stack.pop();
              
          } else if (!stack.isEmpty() && s.charAt(i) == ']' && stack.peek() == '[') {
              stack.pop();
              
          } else if (!stack.isEmpty() && s.charAt(i) == ')' && stack.peek() == '(') {
              stack.pop();
          } else {
              return false;
          }
      }
      
      return stack.isEmpty();
      
  }
}

File: Stacks/Easy/Valid-Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
 var isValid = function(s) {
  let stack = []
  for (let i = 0; i < s.length; i++) {
    const element = stack[stack.length - 1];

    if (s[i] == '{' || s[i] == '[' || s[i] == '(') {
      stack.push(s[i])

    } else if (s[i] == '}' && element == '{' && stack.length !== 0) {
      stack.pop(s[i])
    } else if (s[i] == ')' && element == '(' && stack.length !== 0) {
      stack.pop(s[i])
    } else if (s[i] == ']' && element == '[' && stack.length !== 0) {
      stack.pop(s[i])
    } else {
      return false
    }
    
  }

  return stack.length === 0
};